modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,204 @@
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: A hashing object that contains the cryptographic digest of the HTTP request.
+        :return: A base-64 string representing the cryptographic signature of the input digest.
+        """
+        self.configuration.load_private_key()
+        privkey = self.configuration.private_key
+        if isinstance(privkey, RSA.RsaKey):
+            if self.configuration.signing_algorithm == 'PSS':
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(privkey).sign(digest)
+            elif self.configuration.signing_algorithm == 'PKCS1v15':
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(privkey).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        elif isinstance(privkey, ECC.EccKey):
+            if self.configuration.signing_algorithm in ECDSA_KEY_SIGNING_ALGORITHMS:
+                signature = DSS.new(privkey, self.configuration.signing_algorithm).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        else:
+            raise Exception("Unsupported private key: {0}".format(type(privkey)))
+        return b64encode(signature)
+
+    def get_authorization_header(self, signed_headers, signed_msg):
+        """
+        Calculates and returns the value of the 'Authorization' header when signing HTTP requests.
+        
+        :param signed_headers : A list of strings. Each value is the name of a HTTP header that
+           must be included in the HTTP signature calculation.
+        :param signed_msg: A base-64 encoded string representation of the signature.
+        :return: The string value of the 'Authorization' header, representing the signature
+           of the HTTP request.
+        """
+
+        headers_value = ""
+        created_ts = signed_headers.get('(created)')
+        expires_ts = signed_headers.get('(expires)')
+        lower_keys = [k.lower() for k in signed_headers]
+        headers_value = " ".join(lower_keys)
+
+        auth_str = "Signature keyId=\"{0}\",algorithm=\"{1}\","
+            .format(self.configuration.key_id, self.configuration.signing_scheme)
+        if created_ts is not None:
+            auth_str = auth_str + "created={0},".format(created_ts)
+        if expires_ts is not None:
+            auth_str = auth_str + "expires={0},".format(expires_ts)
+        auth_str = auth_str + "headers=\"{0}\",signature=\"{1}\""
+            .format(headers_value, signed_msg.decode('ascii'))
+        return auth_str
+{{/hasHttpSignatureMethods}}
```

Comment:
This module should only be included if this signature method is used.
So we can enable/disable the inclusion of this module in the Java code.
Because the Java will handle it, can you delete this hasHttpSignatureMethods mustache tag?

Reply:
done